azure/services/loadbalancers/spec_test.go

```diff
@@ -0,0 +1,335 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package loadbalancers
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+)
+
+func getExistingLBWithMissingLBRules() network.LoadBalancer {
+	existingLB := newDefaultPublicAPIServerLB()
+	existingLB.ID = to.StringPtr("azure/my-publiclb")
+	existingLB.LoadBalancingRules = &[]network.LoadBalancingRule{}
+
+	return existingLB
+}
+
+func getExistingLBWithMissingBackendPool() network.LoadBalancer {
+	existingLB := newDefaultPublicAPIServerLB()
+	existingLB.ID = to.StringPtr("azure/my-publiclb")
+	existingLB.BackendAddressPools = &[]network.BackendAddressPool{}
+
+	return existingLB
+}
+
+func getExistingLBWithMissingOutboundRules() network.LoadBalancer {
+	existingLB := newDefaultPublicAPIServerLB()
+	existingLB.ID = to.StringPtr("azure/my-publiclb")
+	existingLB.OutboundRules = &[]network.OutboundRule{}
+
+	return existingLB
+}
+
+func getExistingLBWithMissingProbes() network.LoadBalancer {
+	existingLB := newDefaultPublicAPIServerLB()
+	existingLB.ID = to.StringPtr("azure/my-publiclb")
+	existingLB.Probes = &[]network.Probe{}
+
+	return existingLB
+}
+
+func TestParameters(t *testing.T) {
+	testcases := []struct {
+		name          string
+		spec          *LBSpec
+		existing      interface{}
+		expect        func(g *WithT, result interface{})
+		expectedError string
+	}{
+		{
+			name:     "public API load balancer exists with all expected values",
+			spec:     &fakePublicAPILBSpec,
+			existing: newDefaultPublicAPIServerLB(),
+			expect: func(g *WithT, result interface{}) {
+				g.Expect(result).To(BeNil())
+			},
+			expectedError: "",
+		},
+		{
+			name:     "internal API load balancer with all expected values",
+			spec:     &fakeInternalAPILBSpec,
+			existing: newDefaultInternalAPIServerLB(),
+			expect: func(g *WithT, result interface{}) {
+				g.Expect(result).To(BeNil())
+			},
+			expectedError: "",
+		},
+		{
+			name:     "node outbound load balancer exists with all expected values",
+			spec:     &fakeNodeOutboundLBSpec,
+			existing: newDefaultNodeOutboundLB(),
+			expect: func(g *WithT, result interface{}) {
+				g.Expect(result).To(BeNil())
+			},
+			expectedError: "",
+		},
+		{
+			name:     "load balancer exists with missing load balancing rules",
```

Comment:
If my comment above makes sense, let's check why this test case didn't catch the values being overwritten and fix it so it would have (and make sure it then passes after fixing the issue)

Reply:
I think your comment about it being overwritten makes sense. I believe the reason it wasn't failing was because all the fields are replaced w/ the same values since the spec is still the same.